app/controllers/api/v2/ansible_playbooks_controller.rb

```diff
@@ -0,0 +1,68 @@
+# frozen_string_literal: true
+
+module Api
+  module V2
+    # API controller for Ansible Roles
+    class AnsiblePlaybooksController < ::Api::V2::BaseController
+      include ::Api::Version2
+      include ::ForemanAnsible::ProxyAPI
+
+      before_action :find_proxy, only: [:fetch, :sync]
+
+      resource_description do
+        api_version 'v2'
+        api_base_url '/ansible/api'
+      end
+
+      api :PUT, '/ansible_playbooks/sync', N_('Sync Ansible playbooks')
+      param :proxy_id, :identifier, :required => true, :desc => N_('Smart Proxy to sync from')
+      param :playbooks_names, Array, N_('Ansible  playbooks names to be synced')
+      def sync
+        @task = plan_ansible_sync(@proxy.id, playbooks_names)
+      end
+
+      api :GET, '/ansible_playbooks/fetch', N_('Fetch Ansible playbooks available to be synced')
+      param :proxy_id, :identifier, N_('Smart Proxy to fetch from'),
+            :required => true
+      def fetch
+        fetched = fetch_playbooks_names
+        render :json => { :results => { :playbooks_names => fetched } }
+      end
+
+      def action_permission
+        case params[:action]
+        when 'sync'
+          :sync
+        when 'fetch'
+          :fetch
+        else
+          super
+        end
```

Comment:
I believe this needs to be updated now:
```suggestion
        case params[:action]
        when 'sync', 'fetch'
          :import
        else
          super
        end
```

Reply:
is should be:   ```
     case params[:action]
        when 'sync', 'fetch'
          :sync
        else
          super
        end
``` ? why :import?

Reply:
you've called the permission `import_ansible_playbooks` :) thus it needs to be import for it to map the actions to a correct permission.